idea/idea-completion/src/org/jetbrains/kotlin/idea/completion/handlers/handlerUtils.kt

```diff
@@ -144,8 +146,14 @@ fun createKeywordConstructLookupElement(
                     val beforeCaret = tailBeforeCaret.indentLinesAfterFirst(newIndent)
                     val afterCaret = tailAfterCaret.indentLinesAfterFirst(newIndent)
 
-                    insertionContext.document.insertString(offset, beforeCaret + afterCaret)
-                    insertionContext.editor.moveCaret(offset + beforeCaret.length)
+                    val element = insertionContext.file.findElementAt(offset)?.getNextSiblingIgnoringWhitespaceAndComments(true)
```

Comment:
@goodwinnk Thanks for the review.

`getNextSiblingIgnoringWhitespaceAndComments()` is needed because otherwise the following case will be handled wrongly:

```kotlin
class Foo cons<caret> () //note the space before the primary constructor
```

I agree that comments shouldn't be skipped. For that I could introduce a new extension function `getNextSiblingIgnoringWhitespace()` and use it instead.

Regarding your examples where my logic would lead to arguably wrong behavior: How about I restrict my logic to cases where there isn't a line break between the keyword and the brace/paran?

So this 

```kotlin
fun other(a: Any) {
  if<caret>
  (a as Int).let { println(a + 12) }
}
```

will become

```kotlin
fun other(a: Any) {
  if (<caret>)
  (a as Int).let { println(a + 12) }
}
```

but this

```kotlin
fun other(a: Any) {
  whil<caret> ()
}
```

will become
```kotlin
fun other(a: Any) {
  while (<caret>)
}
```